src/routing/route.cc

```diff
@@ -190,31 +200,164 @@ LaneSRangeRelation Route::ComputeLaneSRangeRelation(const api::LaneSRange& lane_
+  auto prepend_lane_s_range_to_route = [this](const LaneSRangeIndex& lane_s_range_index,
+                                              std::vector<api::LaneSRange>* lane_s_ranges) -> void {
+    lane_s_ranges->insert(lane_s_ranges->begin(), GetLaneSRange(lane_s_range_index.first, lane_s_range_index.second));
   };
 
-  for (size_t i = 0; i < phases_.size(); ++i) {
-    for (size_t j = 0; j < phases_[i].lane_s_ranges().size(); ++j) {
-      if (is_lane_s_range_contained(phases_[i].lane_s_ranges()[j], lane_s_range)) {
-        return {std::make_pair(i, j)};
+  // Returns true when @p lane_s_range_index_a.first is equal to @p lane_s_range_index_b.first.
+  auto is_same_phase = [](const Route::LaneSRangeIndex& lane_s_range_index_a,
+                          const Route::LaneSRangeIndex& lane_s_range_index_b) -> bool {
+    return lane_s_range_index_a.first == lane_s_range_index_b.first;
+  };
+
+  LaneSRangeIndex current_lane_s_range_index = end_lane_s_range_index.value();
+  std::vector<api::LaneSRange> lane_s_ranges;
+
+  // The following double nested loop performs the routing within the Route from the ending api::LaneSRange
+  // towards the starting api::LaneSRange. The is always a possible logical solution, and it is based on
+  // one simple set of premises:
+  //
+  // 1- Move straight backwards first, otherwise
+  // 2- Move towards the left / right, the one that offers the lesser number of hops with a possible
+  //    straight backwards movement. Go to step 1.
+
+  // TODO(#543): this process should be simpler once we process in the constructor end-to-end Phase
+  // connectivity as we can cache in the interfaces those LaneSRangeIndices that connect with each other.
+  while (!is_same_phase(current_lane_s_range_index, *start_lane_s_range_index)) {
+    // Adds the current LaneSRange to the last list.
+    prepend_lane_s_range_to_route(current_lane_s_range_index, &lane_s_ranges);
+    // Moves within the Phase towards the first LaneSRange with a straight predecessor.
+    std::optional<LaneSRangeIndex> predecessor_lane_s_range_index{};
+    while (!predecessor_lane_s_range_index.has_value()) {
+      predecessor_lane_s_range_index = FindStraightPredecessor(current_lane_s_range_index);
+      if (!predecessor_lane_s_range_index.has_value()) {
+        current_lane_s_range_index.second +=
+            FindDirectionTowardsLaneSRangeWithStraightPredecessor(current_lane_s_range_index);
+        prepend_lane_s_range_to_route(current_lane_s_range_index, &lane_s_ranges);
       }
     }
+    // Changes to the preceeding Phase - LaneSRange.
+    current_lane_s_range_index = *predecessor_lane_s_range_index;
+  }
+
+  // TODO(#543): make the set of LaneSRanges of the right length once we can easily map s-coordinates LaneSRange to
+  // LaneSRange.
+  LaneSRangeRelation lane_s_range_relation{LaneSRangeRelation::kUnknown};
+  while (lane_s_range_relation != LaneSRangeRelation::kCoincident) {
```

Comment:
Can you add a comment explaining what this while loop does? I am having trouble following it.

Reply:
Added. Also,

![image](https://github.com/maliput/maliput/assets/3825465/d7e3ee12-7410-4716-98e0-c9e02143f389)

This does the actual jump from D to E in the path above. Note that the to move from A to D we used the previous loop but we could not actually move within the starting Phase towards the starting api::LaneSRange.